File: Leetcode String/151. Reverse Words in a String.cpp
class Solution {
public:
    string reverseWords(string s) {
        string ans="";
        string temp="";
        for(int i=s.length()-1; i>=0; i--){
            if(s[i]== ' ' && temp.length()!= 0){
                reverse(temp.begin(),temp.end());
                ans=ans+temp;
                ans.push_back(' ');
                temp="";
            }
            else if(s[i] != ' ')
                temp.push_back(s[i]);
        }
         reverse(temp.begin(),temp.end());
                ans=ans+temp;
            if(ans[ans.length()-1] == ' ')
                ans.pop_back();
                return ans;
    }
};

File: java/leetcode java/151. Reverse Words in a String.java
class Solution {
    public String reverseWords(String s) {
        if(s == null)
            return null;
        String str[]=s.trim().split("\\s+");
        int st=0;
        int l=str.length-1;
        while(st<l){
            String temp=str[st];
            str[st]=str[l];
            str[l]=temp;
            st++;
            l--;
        }
        return String.join(" ",str);
    }
}